schema_utils.py

```diff
@@ -387,3 +388,61 @@ def is_valid_email(obj):
             bool. Whether the given object is a valid email.
         """
         return bool(re.search(r'^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$', obj))
+
+    @staticmethod
+    def is_valid_math_expression(obj, algebraic=True):
+        """Checks if the given obj (a string) represents a valid algebraic or
+        numeric expression.
+
+        Args:
+            obj: str. The given expression.
+            algebraic: bool. True if the given expression is algebraic
+                else numeric.
+
+        Returns:
+            bool. Whether the given object is a valid expression.
+        """
+
+        if not expression_parser.is_valid_expression(obj):
+            return False
+
+        expression_is_algebraic = expression_parser.is_algebraic(obj)
+        # If the algebraic flag is true, expression_is_algebraic should
+        # also be true, otherwise both should be false which would imply
+        # that the expression is numeric.
+        return not algebraic ^ expression_is_algebraic
+
+    @staticmethod
+    def is_valid_math_equation(obj):
+        """Checks if the given  obj (a string) represents a valid math equation.
+
+        Args:
+            obj: str. A string.
+
+        Returns:
+            bool. Whether the given object is a valid math equation.
+        """
+        if obj.count('=') != 1:
+            return False
+
+        is_valid_math_expression = get_validator(
+            'is_valid_math_expression')
+        lhs, rhs = obj.split('=')
+
+        # Both sides have to be valid expressions and at least one of them has
+        # to be a valid algebraic expression.
+        lhs_is_algebraically_valid = is_valid_math_expression(lhs)
+        rhs_is_algebraically_valid = is_valid_math_expression(rhs)
+
+        lhs_is_numerically_valid = is_valid_math_expression(
```

Comment:
Wouldn't algebraic validity imply numerical validity, since we are removing the restriction that at least one token should be an identifier?

Reply:
No, it wouldn't. The restriction is still present. For algebraic expressions, at least one token should be a valid identifier and for numeric, there should be none. Could you please point towards the discussion/code that caused this confusion so that I can fix it?

Reply:
Oh, ok, thanks for clarifying! You could add a comment above the XOR line in `is_valid_math_expression`, other than that I think it's clear enough.

Reply:
I've added a comment in the docstring of that function that should clear things up.